slides/3-foundations-of-javascript.html

```diff
@@ -388,31 +395,113 @@ <h2><i class="fab fa-js"></i> Foundations of JavaScript</h2>
                         </textarea>
                     </section>
 
-                    <section data-markdown style="font-size: 0.7em;">
+                    <section data-markdown style="font-size: 0.65em;">
                         <textarea data-template>
                             ## <i class="fab fa-js"></i> JavaScript's Types
 
                             ECMAScript defines 7 **primitive** (Immutable) types for values:
+
+                            - `Undefined`: Unique value `undefined`
+                            - `Number`: Real or integer number (e.g. `3.14`, `42`)
+                            - `Boolean`: `true` or `false`
+                            - `String`: Character sequence, whose literals begin and end with single or double quotes (e.g. `"HEIG-VD"`, `'hello'`)
+                            - `BigIng`: Arbitrary-precision integers, whose literals end with an `n` (e.g. `9007199254740992n`)
+                            - `Symbol`: Grobally unique values usable as identifiers or keys in objects (e.g. `Symbol()`, `Symbol("description")`)
+                            - `Null`: Unique value `null`
+
+                            In a dynamic language you don't specify the type when you declare a variable and the type of a variable can change.
                             
+                            ECMAScript also defines a special mutable type called **object** for collections of properties (objects and array).
+
+                            https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Data_types
+
+                            Notes:
+
+                            The `Symbol` data type is special.
+
+                            Calling `Symbol("description")` creates a symbol
+                            value with description `"description"`.  That value
+                            is guaranteed to be unique in the entire execution
+                            context (hence even across different libraries or
+                            modules). Therefore, two calls to `Symbol("desc")`
+                            will return two different values.
+
+                            It can be used anywhere an identifier or a key is
+                            expected, so for example an object `car` (see next
+                            slide) could have a property whose index is
+                            `Symbol("model")`, rather than simply
+                            `"model"`. This would ensure that this property does
+                            not get overridden by mistake by an inheritting
+                            class (see next chapter on OOP), for example.
+
+                        </textarea>
+                    </section>
+
+                    <section data-markdown style="font-size: 0.75em;">
+                        <textarea data-template>
+                            ## <i class="fab fa-js"></i> Objects
```

Comment:
Mentioning objects and arrays here already, just the basics, since it will be used later in this slide deck, before the next chapter on OOP where it really gets introduced.